src/main/webapp/app/core/main/controllers/createSaleDialogController.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+(function () {
+    var app = angular.module("efApp");
+
+    app.controller("CreateSaleDialogController", function CreateSaleDialogController(sales, toastr, 
+        ProductService, SalesService, $uibModalInstance) {
+        var createSaleCtrl = this;
+        createSaleCtrl.products = [];
+        createSaleCtrl.saleProducts = [];
+
+        createSaleCtrl.createSale = function createSale() {
+            var sale = prepareSale();
```

Comment:
var sale = createSaleCtrl.prepareSale();
would not that be correct?

Reply:
prepareSale is a private function